libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -405,6 +452,18 @@ def _handle_openai_bad_request(e: openai.BadRequestError) -> None:
         raise
 
 
+def _is_builtin_tool(tool: dict) -> bool:
+    return "type" in tool and tool["type"] != "function"
+
+
+def _transform_payload_for_responses(payload: dict) -> dict:
+    updated_payload = payload.copy()
+    if messages := updated_payload.pop("messages"):
+        updated_payload["input"] = messages
```

Comment:
think we need to upload contents here

Reply:
also think function tools look a lil different, should probably check for it. they unnested
from {"type": "function", "function": {"name": ...} to {"type": "function", "name": ...}